plugin/storage/es/factory.go

```diff
@@ -317,9 +308,40 @@ func (f *Factory) CreateSamplingStore(maxBuckets int) (samplingstore.Store, erro
 		Lookback:               f.primaryConfig.AdaptiveSamplingLookback,
 		MaxDocCount:            f.primaryConfig.MaxDocCount,
 	})
+
+	if f.primaryConfig.CreateIndexTemplates && !f.primaryConfig.UseILM {
+		mappingBuilder := mappingBuilderFromConfig(f.primaryConfig)
+		samplingMapping, err := mappingBuilder.GetSamplingMappings()
+		if err != nil {
+			return nil, err
+		}
+		primaryClient := f.getPrimaryClient()
+		if primaryClient == nil {
+			return nil, errors.New("primary client is nil")
+		}
+		templateId := samplingTemplatePrefix + time.Now().UTC().Format(f.primaryConfig.IndexDateLayoutSampling)
```

Comment:
this is not an acceptable hack. None of the other code does this, e.g. in the Writer:
```
	_, err := s.client().CreateTemplate(indexPrefix + "jaeger-span").Body(spanTemplate).Do(context.Background())
```
The template names are supposed to be stable, fixed names, not randomized strings.

Reply:
I updated to use a fixed name and added a prefix to the sampling template name.
8118847da8567f9b95257740bb91213299022667
04ca6ff4033a8f1b2c54909c6259ff1383aeb64d